File: client/src/app/models/order.ts
import { IAddress } from "./address";
import { IDeliveryMethod } from "./deliveryMethod";

export interface IOrderToCreate {
  basketId: string;
  deliveryMethodId: number;
  shipToAddress: IAddress
}

export class OrderListParam {
  sort: string = '';
  pageIndex: number = 1;
  pageSize: number = 10;
}

export interface IOrder {
  id: number
  buyerName: string,
  buyerEmail: string
  totalPrice: number
  orderDate: string
  orderAddress: IAddress
  deliveryMethod: IDeliveryMethod
  orderItems: IOrderItem[]
  orderStatus: string
  paymentIntentId: string
}
export interface IOrderItem {
  id: number
  name: string
  description: string
  price: number
  imgUrl: string
  langCode: string
  quantity: number
  createdDate: string
}

File: backendApi/BackendApi/Core/Models/Entities/Order.cs
using System;
using System.Collections.Generic;


namespace BackendApi.Core.Models.Entities
{
    public class Order : IBaseEntity
    {
        public Order()
        {
            
        }
        public Order(
            IReadOnlyList<OrderItem> orderItems,
            string buyerName,
            string buyerEmail,
            OrderAddress orderAddress,
            int deliveryMethodId,
            decimal totalPrice,
            string paymentIntentId
           )
        {
            OrderItems = orderItems;
            BuyerName = buyerName;
            BuyerEmail = buyerEmail;
            TotalPrice = totalPrice;
            OrderAddress = orderAddress;
            DeliveryMethodId = deliveryMethodId;
            PaymentIntentId = paymentIntentId;
        }
        public string BuyerName  { get; set; }
        public string BuyerEmail { get; set; }
        public decimal TotalPrice { get; set; }
        public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.UtcNow;
        public OrderAddress OrderAddress { get; set; }
        public int OrderAddressId { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public int DeliveryMethodId { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }
        public int Id { get ; set ; }
        public DateTimeOffset CreatedDate { get ; set ; }
        public DateTimeOffset? ModifiedDate { get ; set ; }
    }
}